modules/gapi/samples/onevpl_infer_single_roi.cpp

```diff
@@ -316,22 +401,53 @@ int main(int argc, char *argv[]) {
         device_id
     };
 
-    // Create device_ptr & context_ptr using graphic API
-    // InferenceEngine requires such device & context to create its own
-    // remote shared context through InferenceEngine::ParamMap in
-    // GAPI InferenceEngine backend to provide interoperability with onevpl::GSource
-    // So GAPI InferenceEngine backend and onevpl::GSource MUST share the same
-    // device and context
-    cv::util::optional<cv::gapi::wip::onevpl::Device> accel_device;
-    cv::util::optional<cv::gapi::wip::onevpl::Context> accel_ctx;
-
+    // It is allowed to reuse predefined device_ptr & context_ptr received
+    // from user application. Current sample demonstrate how to deal with this situation.
+    //
+    // But if you do not need this fine-grained acceleration devices configuration then
+    // just use default constructors for onevpl::GSource, IE and preprocessing module.
+    // But please pay attention that default pipeline construction in this case would be
+    // very inefficient and carries out multiple CPU-GPU memory copies
+    //
+    // If you want to reach max performance and apply copy-less approach for specific
+    // device & context selection then follow the steps below.
+    // The situation is complicated a little bit:
+    //
+    // - all component-participants (Source, Preprocessing, Inference)
+    // must share the same Device & Context
+    //
+    // - wrap you available device & context into `cv::gapi::wip::Device` &
+    // `cv::gapi::wip::Context`
+    //
+    // - pass such wrappers as constructor arguments for each component in pipeline:
+    //      a) use special constructor for `onevpl::GSource`
+    //      b) use `cfgContextParams` method of `cv::gapi::ie::Params` to charge PreprocesingEngine
+    //      c) use `InferenceEngine::ParamMap` to activate remote ctx in Inference Engine
```

Comment:
Looks like it will be very nice to have an opportunity to configure everything in one place (set device/context only once) if user want everything to run on GPU for example. But can't find a way how to make it, at first glance

Reply:
Indeed, that's because we discussed DeviceSelector approach many times before